### PR TITLE
PIL-148 CYA page (Repayment + Non UK bank details)

### PR DESCRIPTION
**Pull request by javed-iqbal1**

### CODE REVIEW

File: app/views/repayments/RepaymentsCheckYourAnswersView.scala.html [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * limitations under the License.
+ *@
+
+@import config.FrontendAppConfig
+@import views.html.components.gds._
+
+@this(
+        layout: templates.Layout,
+        govukSummaryList: GovukSummaryList,
+        sectionHeader: sectionHeader,
```
Comment (jassalrichy) [inline app/views/repayments/RepaymentsCheckYourAnswersView.scala.html:23]: remove unused sectionHeader ID

File: app/viewmodels/checkAnswers/repayments/RepaymentsContactNameSummary.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.checkAnswers.repayments
+
+import models.{CheckMode, UserAnswers}
+import pages.{RepaymentsContactNamePage, RepaymentsRefundAmountPage}
```
Comment (jassalrichy) [inline app/viewmodels/checkAnswers/repayments/RepaymentsContactNameSummary.scala:20]: remove unused import RepaymentsRefundAmountPage

File: app/viewmodels/checkAnswers/repayments/RepaymentsContactEmailSummary.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.checkAnswers.repayments
+
+import models.{CheckMode, UserAnswers}
+import pages.{RepaymentsContactEmailPage, RepaymentsRefundAmountPage}
```
Comment (jassalrichy) [inline app/viewmodels/checkAnswers/repayments/RepaymentsContactEmailSummary.scala:20]: remove unused import RepaymentsRefundAmountPage

File: app/viewmodels/checkAnswers/repayments/RepaymentsContactByTelephoneSummary.scala [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.checkAnswers.repayments
+
+import models.{CheckMode, UserAnswers}
+import pages.{RepaymentsContactByTelephonePage, RfmContactByTelephonePage}
```
Comment (jassalrichy) [inline app/viewmodels/checkAnswers/repayments/RepaymentsContactByTelephoneSummary.scala:20]: remove unused import RfmContactByTelephonePage

File: app/viewmodels/checkAnswers/repayments/RepaymentsTelephoneDetailsSummary.scala [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.checkAnswers.repayments
+
+import models.{CheckMode, UserAnswers}
+import pages.{RepaymentsTelephoneDetailsPage, RfmCapturePrimaryTelephonePage}
```
Comment (jassalrichy) [inline app/viewmodels/checkAnswers/repayments/RepaymentsTelephoneDetailsSummary.scala:20]: remove unused import RfmCapturePrimaryTelephonePage

File: conf/messages.en [hunk 6]
```diff
@@ -1276,6 +1333,28 @@ ukOrAbroadBankAccount.title = What type of account will the refund be sent to?
 ukOrAbroadBankAccount.heading = What type of account will the refund be sent to?
 ukOrAbroadBankAccount.ukBankAccount = UK bank account
 ukOrAbroadBankAccount.nonUkBankAccount = Non-UK bank account
-ukOrAbroadBankAccount.checkYourAnswersLabel = UkOrAbroadRepayment
+ukOrAbroadBankAccount.checkYourAnswersLabel = What type of account will the refund be sent to?
 ukOrAbroadBankAccount.error.required = Select what type of account the refund will be sent to
-ukOrAbroadBankAccount.change.hidden = UkOrAbroadBankAccount
+ukOrAbroadBankAccount.change.hidden = the type of account
```
Comment (jassalrichy) [inline conf/messages.en:1338]: Change:
ukOrAbroadBankAccount.change.hidden = the type of account
To:
ukOrAbroadBankAccount.change.hidden = what type of account will the refund be sent to?

File: app/viewmodels/checkAnswers/repayments/NonUKBankNameSummary.scala [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.checkAnswers.repayments
+
+import models.{CheckMode, UserAnswers}
+import pages.{FmRegisteredAddressPage, NonUKBankPage}
```
Comment (jassalrichy) [inline app/viewmodels/checkAnswers/repayments/NonUKBankNameSummary.scala:20]: remove unused imports: FmRegisteredAddressPage & CountryOptions

File: conf/messages.en [hunk 8]
```diff
@@ -1246,6 +1269,40 @@ repayments.nonUKBank.error.bic.length = BIC or SWIFT code must be between 8 and
 repayments.nonUKBank.error.iban.length = IBAN must be up to 34 characters
 repayments.nonUKBank.error.bic.format = Enter a valid BIC or SWIFT code like HBUKGB4B
 repayments.nonUKBank.error.iban.format = Enter a valid IBAN like GB29NWBK60161331926819
+repayments.nonUKBank.summary.bankName.checkYourAnswersLabel = Name of the bank
+repayments.nonUKBank.summary.bankName.checkYourAnswersLabel.hidden =  the name of the bank
+repayments.nonUKBank.summary.nameOnAccount.checkYourAnswersLabel = Name on account
+repayments.nonUKBank.summary.nameOnAccount.checkYourAnswersLabel.hidden =  the name on account
```
Comment (jassalrichy) [inline conf/messages.en:1275]: Change:
repayments.nonUKBank.summary.nameOnAccount.checkYourAnswersLabel.hidden = the name on account
To:
repayments.nonUKBank.summary.nameOnAccount.checkYourAnswersLabel.hidden = the name on bank account

File: app/viewmodels/checkAnswers/repayments/RequestRefundAmountSummary.scala [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.checkAnswers.repayments
+
+import models.{CheckMode, UserAnswers}
+import pages.{RepaymentsRefundAmountPage, UpeNameRegistrationPage}
```
Comment (jassalrichy) [inline app/viewmodels/checkAnswers/repayments/RequestRefundAmountSummary.scala:20]: remove unused import UpeNameRegistrationPage

File: app/navigation/RepaymentNavigator.scala [hunk 10]
```diff
@@ -52,6 +57,38 @@ class RepaymentNavigator @Inject() {
       }
       .getOrElse(routes.JourneyRecoveryController.onPageLoad())
 
-  private val checkRouteMap: Page => Option[String] => UserAnswers => Call = _ => _ => _ => routes.IndexController.onPageLoad
+  private val checkRouteMap: Page => Option[String] => UserAnswers => Call = {
+    case RepaymentsRefundAmountPage       => id => _ => controllers.repayments.routes.RepaymentsCheckYourAnswersController.onPageLoad(id)
+    case ReasonForRequestingRefundPage    => id => _ => controllers.repayments.routes.RepaymentsCheckYourAnswersController.onPageLoad(id)
+    case UkOrAbroadBankAccountPage        => id => _ => controllers.repayments.routes.RepaymentsCheckYourAnswersController.onPageLoad(id)
```
Comment (jassalrichy) [inline app/navigation/RepaymentNavigator.scala:63]: Need to add logic for correct routing when the user changes value of UkOrAbroadBankAccountPage.

File: app/navigation/RepaymentNavigator.scala [hunk 11]
```diff
@@ -52,6 +57,38 @@ class RepaymentNavigator @Inject() {
...
+      .get(RepaymentsContactByTelephonePage)
+      .map {
+        case true =>
+          controllers.repayments.routes.RepaymentsTelephoneDetailsController.onPageLoad(clientPillar2Id = maybeClientId, NormalMode)
+        case false =>
+          controllers.repayments.routes.RepaymentsCheckYourAnswersController.onPageLoad(clientPillar2Id = maybeClientId)
+      }
+      .getOrElse(routes.JourneyRecoveryController.onPageLoad())
+
+  private def telephonePreferenceCheckMode(maybeClientId: Option[String], userAnswers: UserAnswers): Call =
```
Comment (jassalrichy) [inline app/navigation/RepaymentNavigator.scala:83]: If user selects to change TelephonePreference, and make no amend then they should be redirected back to the CYA page and not to the page that captures the telephone number.

File: app/viewmodels/checkAnswers/repayments/ReasonForRequestingRefundSummary.scala [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.checkAnswers.repayments
+
+import models.{CheckMode, UserAnswers}
+import pages.{ReasonForRequestingRefundPage, SubPrimaryContactNamePage}
```
Comment (jassalrichy) [inline app/viewmodels/checkAnswers/repayments/ReasonForRequestingRefundSummary.scala:20]: Remove unused import SubPrimaryContactNamePage

File: app/navigation/RepaymentNavigator.scala [hunk 13]
```diff
@@ -52,6 +57,38 @@ class RepaymentNavigator @Inject() {
...
+    userAnswers
+      .get(RepaymentsContactByTelephonePage)
+      .map {
+        case true =>
+          controllers.repayments.routes.RepaymentsTelephoneDetailsController.onPageLoad(clientPillar2Id = maybeClientId, CheckMode)
+        case false =>
+          controllers.repayments.routes.RepaymentsCheckYourAnswersController.onPageLoad(clientPillar2Id = maybeClientId)
+      }
+      .getOrElse(routes.JourneyRecoveryController.onPageLoad())
```
Comment (jassalrichy) [inline app/navigation/RepaymentNavigator.scala:92]: Need a test for redirecting to journey recovery.

File: app/navigation/RepaymentNavigator.scala [hunk 14]
```diff
@@ -52,6 +57,38 @@ class RepaymentNavigator @Inject() {
...
+  private def telephonePreferenceNormalMode(maybeClientId: Option[String], userAnswers: UserAnswers): Call =
+    userAnswers
+      .get(RepaymentsContactByTelephonePage)
+      .map {
+        case true =>
+          controllers.repayments.routes.RepaymentsTelephoneDetailsController.onPageLoad(clientPillar2Id = maybeClientId, NormalMode)
+        case false =>
+          controllers.repayments.routes.RepaymentsCheckYourAnswersController.onPageLoad(clientPillar2Id = maybeClientId)
+      }
+      .getOrElse(routes.JourneyRecoveryController.onPageLoad())
```
Comment (jassalrichy) [inline app/navigation/RepaymentNavigator.scala:81]: Need a test for redirecting to journey recovery.

File: test/views/repayments/RepaymentsCheckYourAnswersViewSpec.scala [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+import pages.{NonUKBankPage, ReasonForRequestingRefundPage, RepaymentsContactByTelephonePage, RepaymentsContactEmailPage, RepaymentsContactNamePage, RepaymentsRefundAmountPage, RepaymentsTelephoneDetailsPage, RfmNameRegistrationPage, RfmRegisteredAddressPage, UkOrAbroadBankAccountPage}
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryList
+import viewmodels.checkAnswers.repayments.{NonUKBankBicOrSwiftCodeSummary, NonUKBankIbanSummary, NonUKBankNameOnAccountSummary, NonUKBankNameSummary, ReasonForRequestingRefundSummary, RepaymentsContactByTelephoneSummary, RepaymentsContactEmailSummary, RepaymentsContactNameSummary, RepaymentsTelephoneDetailsSummary, RequestRefundAmountSummary, UkOrAbroadBankAccountSummary}
+import viewmodels.checkAnswers.{RfmNameRegistrationSummary, RfmRegisteredAddressSummary}
+import viewmodels.govuk.summarylist._
+import views.html.repayments.RepaymentsCheckYourAnswersView
+import views.html.rfm.RfmCheckYourAnswersView
+
+class RepaymentsCheckYourAnswersViewSpec extends ViewSpecBase {
+  val amount = BigDecimal(9.99)
```
Comment (jassalrichy) [inline test/views/repayments/RepaymentsCheckYourAnswersViewSpec.scala:34]: Add type annotation.

File: app/navigation/RepaymentNavigator.scala (hunk 11 above)
Comment (javed-iqbal1) [reply app/navigation/RepaymentNavigator.scala:83]: check in place already

File: app/navigation/RepaymentNavigator.scala (hunk 10 above)
Comment (javed-iqbal1) [reply app/navigation/RepaymentNavigator.scala:63]: once other screen will be ready we can add that logic

**Review by jassalrichy — DISMISSED** [review]

**Review by jassalrichy — DISMISSED** [review]

File: test/controllers/UnderConstructionControllerSpec.scala [hunk 16]
```diff
@@ -40,5 +40,21 @@ class UnderConstructionControllerSpec extends SpecBase {
         contentAsString(result) mustEqual view()(request, appConfig(application), messages(application)).toString
       }
     }
+
+    "must return OK and the correct view for a GET for repayments" in {
```
Comment (jassalrichy) [inline test/controllers/UnderConstructionControllerSpec.scala:44]: ... GET for Agent

File: app/viewmodels/checkAnswers/repayments/RequestRefundAmountSummary.scala [hunk 17]
```diff
@@ -30,7 +30,7 @@ object RequestRefundAmountSummary {
     answers.get(RepaymentsRefundAmountPage).map { answer =>
       SummaryListRowViewModel(
         key = "requestRefundAmount.checkYourAnswersLabel",
-        value = ValueViewModel(HtmlFormat.escape("£" + answer.setScale(2)).toString),
+        value = ValueViewModel(HtmlFormat.escape("£" + answer.bigDecimal.setScale(2)).toString.replace(".00", "")),
```
Comment (jassalrichy) [inline app/viewmodels/checkAnswers/repayments/RequestRefundAmountSummary.scala:33]: Is there a reason for removing the decimal places when they are zero ? What if the value is 100.50 we still display the deciaml palces ?

Comment (javed-iqbal1) [reply app/viewmodels/checkAnswers/repayments/RequestRefundAmountSummary.scala:33]: yes it is requirement if decimal part is .00 , will work fine for .50

**Review by jassalrichy — APPROVED** [review]

**Pull request merged**